eng/pipelines/templates/stages/archetype-ios-release.yml

```diff
@@ -0,0 +1,78 @@
+parameters:
+  - name: DependsOn
+    type: object
+  - name: ServiceDirectory
+    type: string
+  - name: Artifacts
+    type: object
+    default: []
+
+stages:
+
+  # We generate two interdepdent stages for each artifact listed in the ci.yml file, creates the release
+  # in GitHub. The Release stage publishes to Maven Central. Both stages require approval since they
+  # effectively burn the version number. For testing of packages prior to burning the version number -
+  # the Validation step below publishes a package to a "burner" feed which is cleaned up after the
+  # pipeline completes.
+  - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
+    - ${{ each artifact in parameters.Artifacts }}:
+      - stage: Release_${{artifact.safeName}}
+        displayName: 'Release: ${{artifact.name}}'
+        dependsOn: ${{parameters.DependsOn}}
+        condition: and(succeeded(), ne(variables['SetDevVersion'], 'true'), ne(variables['Skip.Release'], 'true'), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-java-pr'))
+        jobs:
+          - deployment: TagRepository
+            displayName: "Create release tag"
+            condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
+            environment: github
+            variables:
+              - template: ../variables/globals.yml
+            pool:
+              name: Azure Pipelines
+              vmImage: vs2017-win2016
+            strategy:
+              runOnce:
+                deploy:
+                  steps:
+                    - checkout: self
+                    - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                    # TODO: Will need to be adapted to work for Cocoapods (we'll read versions from the Cocoapod version specs)
+                    # - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
+                    #   parameters:
+                    #     ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
+                    #     PackageRepository: Maven
+                    #     ReleaseSha: $(Build.SourceVersion)
```

Comment:
To get a package version, you can use:
`python3 eng/scripts/version.py check AzureFoo` and it will output the version to STDOUT. We run version verify in CI to ensure that the Xcode projects, podspecs, etc are consistent for each service. 

We probably DO need to verify that a version we intend to push isn't already on CocoaPods, since in theory one could trigger a release build without incrementing the version in the Azure SDK repo.

Reply:
Thanks. I will be implementing this in an upcoming PR.